street-explorer/www/js/lane_editor.js

```diff
@@ -0,0 +1,205 @@
+import { downloadGeneratedFile } from "./files.js";
+import {
+  makeLaneMarkingsLayer,
+  lanePolygonStyle,
+  makePlainGeoJsonLayer,
+} from "./layers.js";
+import { setupLeafletMap } from "./leaflet.js";
+import init, { JsStreetNetwork } from "./osm2streets-js/osm2streets_js.js";
+
+await init();
+
+export class LaneEditor {
+  constructor(mapContainer) {
+    this.map = setupLeafletMap(mapContainer);
+    this.network = null;
+    this.layers = [];
+    this.currentWay = null;
+    this.currentWaysLayer = null;
+
+    // Wire up the import button
+    const importButton = document.getElementById("import-view");
+    importButton.onclick = async () => {
+      if (this.map.getZoom() < 15) {
+        window.alert("Zoom in more to import");
+        return;
+      }
+
+      await this.importView(importButton);
+    };
+  }
+
+  async importView(importButton) {
```

Comment:
Some of this code is duplicated, but the entire thing in street-explorer isn't appropriate. Maybe the overpass fetch could be a helper. Thinking in terms of components, an "import current view" button could be common, providing a callback / event for "here's some XML fetched"